src/app/Fake.DotNet.ILMerge/ILMerge.fs

```diff
@@ -0,0 +1,183 @@
+          TargetKind = Library
+          UnionMerge = false
+          XmlDocs = false }
+
+/// Builds the arguments for the ILMerge task
+/// [omit]
+let internal getArguments outputFile primaryAssembly parameters =
+    let Item a x =
+        if x |> String.IsNullOrWhiteSpace then []
+        else [ sprintf a x ]
+
+    let ItemList a x =
+        if x |> isNull then []
+        else
+            x
+            |> Seq.collect (fun i -> [ sprintf a i ])
+            |> Seq.toList
+
+    let Flag a predicate =
+        if predicate then [ a ]
+        else []
+
+    [ Item "/out:%s" outputFile
+      Item "/ver:%s" (match parameters.Version with
+                      | Some v -> v.ToString()
+                      | None -> String.Empty)
+      Item "/attr:%s" parameters.AttributeFile
+      Item "/keyfile:%s" parameters.KeyFile
+      Item "/log:%s" parameters.LogFile
+      Item "/target:%s" <| (sprintf "%A" parameters.TargetKind).ToLower()
+      Item "/targetplatform:%s" parameters.TargetPlatform
+      Item "/align:%s" (match parameters.FileAlignment with
+                        | Some i -> i.ToString(CultureInfo.InvariantCulture)
+                        | None -> String.Empty)
+      (match parameters.Internalize with
+       | NoInternalize -> []
+       | Internalize -> Flag "/internalize" true
+       | InternalizeExcept excludeFile -> Item "/internalize:%s" excludeFile)
+      Flag "/allowMultiple" parameters.AllowMultipleAssemblyLevelAttributes
+      Flag "/wildcards" parameters.AllowWildcards
+      Flag "/zeroPeKind" parameters.AllowZeroPeKind
+      Flag "/closed" parameters.Closed
+      Flag "/copyattrs" parameters.CopyAttributes
+      Flag "/union" parameters.UnionMerge
+      Flag "/ndebug" (not parameters.DebugInfo)
+      Flag "/xmldocs" parameters.XmlDocs
+      (match parameters.AllowDuplicateTypes with
+       | NoDuplicateTypes -> []
+       | AllPublicTypes -> Flag "/allowDup" true
+       | DuplicateTypes types -> ItemList "/allowDup:%s" types)
+      ItemList "/lib:%s" parameters.SearchDirectories
+      (primaryAssembly :: (parameters.Libraries |> Seq.toList)) ]
+    |> List.concat
+
+/// Builds the process to run for the ILMerge task
+/// [omit]
+
+let internal createProcess parameters outputFile primaryAssembly =
+    let args = getArguments outputFile primaryAssembly parameters
+    CreateProcess.fromRawCommand parameters.ToolPath args
+
+/// Uses ILMerge to merge .NET assemblies.
+/// ## Parameters
+///
+///  - `parameters` - An ILMerge.Params value with your required settings.
+///  - `outputFile` - Output file path for the merged assembly.
+///  - `primaryAssembly` - The assembly you want ILMerge to consider as the primary.
+let run parameters outputFile primaryAssembly =
+    use __ = Trace.traceTask "ILMerge" primaryAssembly
+  
+    // The type initializer for 'System.Compiler.CoreSystemTypes' throws on Mono.
+    // So let task be a no-op marked as failed on non-Windows platforms  
```

Comment:
Or we throw an exception in order to notify the user
`raise <| NotSupportedException("ILMerge is currently not supported on mono")